includes/Checker/Check_Runner.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Interface WordPress\Plugin_Check\Checker\Check_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for runner classes like AJAX runner or CLI runner.
+ *
+ * @since n.e.x.t
+ */
+interface Check_Runner {
+
+	/**
+	 * Determine if the current request is intended for the plugin checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean Returns true if the check is for plugin else false.
+	 */
+	public function is_plugin_check();
+
+	/**
+	 * Handles running of the universal preparations depending on which checks requested.
```

Comment:
@vishalkakadiya a minor edit here.

Additionally, I think we should include doc comments on the callable returned by the method and that it can throw exceptions. See the docblock for the [`Use_Minimal_Theme_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Use_Minimal_Theme_Preparation.php#L52-L60) as an example.

```suggestion
	 * Handles running the universal preparations depending on the requested checks.
```